go/teams/transactions.go

```diff
@@ -0,0 +1,495 @@
+	admin, err := team.getAdminPermission(ctx, true)
+	if err != nil {
+		return err
+	}
+
+	var sections []SCTeamSection
+	memSet := newMemberSet()
+
+	// Transform payloads to SCTeamSections.
+	for _, p := range tx.payloads {
+		section := SCTeamSection{
+			ID:       SCTeamID(team.ID),
+			Admin:    admin,
+			Implicit: team.IsImplicit(),
+			Public:   team.IsPublic(),
+		}
+
+		switch payload := p.(type) {
+		case *keybase1.TeamChangeReq:
+			// We need memberSet for this particular payload, but also keep a
+			// memberSet for entire transaction to generate boxes afterwards.
+			payloadMemberSet, err := newMemberSetChange(ctx, g, *payload)
+			if err != nil {
+				return err
+			}
+
+			// TODO: Instead of loading members twice, have a "append"
+			// function in memberSet.
+			if err := memSet.loadMembers(ctx, g, *payload, true /* forcePoll */); err != nil {
+				return err
+			}
+
+			section.Members, err = payloadMemberSet.Section()
+			if err != nil {
+				return err
+			}
+
+			section.CompletedInvites = payload.CompletedInvites
+			sections = append(sections, section)
+		case *SCTeamInvites:
+			entropy, err := makeSCTeamEntropy()
+			if err != nil {
+				return err
+			}
+
+			section.Invites = payload
+			section.Entropy = entropy
+			sections = append(sections, section)
+		default:
+			return fmt.Errorf("Unhandled case in AddMemberTx.Post, unknown type: %T", p)
+		}
+	}
+
+	// If memSet has any downgrades, request downgrade lease.
+	var merkleRoot *libkb.MerkleRoot
+	var lease *libkb.Lease
+
+	downgrades, err := team.getDowngradedUsers(ctx, memSet)
+	if len(downgrades) != 0 {
+		lease, merkleRoot, err = libkb.RequestDowngradeLeaseByTeam(ctx, g, team.ID, downgrades)
+		if err != nil {
+			return err
+		}
+		// Always cancel lease so we don't leave any hanging.
+		defer func() {
+			err := libkb.CancelDowngradeLease(ctx, g, lease.LeaseID)
+			if err != nil {
+				g.Log.CWarningf(ctx, "Failed to cancel downgrade lease: %s", err.Error())
+			}
+		}()
+	}
+
+	secretBoxes, implicitAdminBoxes, perTeamKeySection, err := team.recipientBoxes(ctx, memSet)
+	if err != nil {
+		return err
+	}
+
+	if perTeamKeySection != nil {
+		// We have a new per team key, find first TeamChangeReq
+		// section created and add perTeamKeySection.
+		// TODO: Should it be the first ChangeMembership sig or the last?
+		for i, v := range tx.payloads {
+			if _, ok := v.(*keybase1.TeamChangeReq); ok {
+				sections[i].PerTeamKey = perTeamKeySection
+				break
+			}
+		}
+	}
+
+	// Take payloads and team sections and generate chain of signatures.
+	nextSeqno := team.NextSeqno()
+	latestLinkID := team.chain().GetLatestLinkID()
+
+	var readySigs []libkb.SigMultiItem
+	for i, section := range sections {
+		var linkType libkb.LinkType
+		switch tx.payloads[i].(type) {
+		case *keybase1.TeamChangeReq:
+			linkType = libkb.LinkTypeChangeMembership
+		case *SCTeamInvites:
+			linkType = libkb.LinkTypeInvite
+		}
```

Comment:
switch needs a default case

Reply:
it would have errored out in the first loop through payload, but I will add case here either way